### DISCUSSION TITLE
How to change the peridynamic weight function?

### DISCUSSION THREAD

**Discussion by andrisfreimanis** [idaholab/moose#26386] [Q&A General] [answered]
Dear All,
I want to use the small strain H1NOSPD with a different weight function. Currently, the weight function is hard coded as `weight_nb = _horizon_radius[_qp] / origin_vec_nb.norm();`, and it appears in the loop calculating the shape and deformation tensor and in _multi[_qp].

I changed it to `std::pow(_horizon_radius[_qp] / origin_vec_nb.norm(), 2.0);` and the Jacobian tests failed. I expected that failure and strain tests would fail as their results are compared against gold files, which have a different weight function. But why would Jacobian tests fail? As I understand, in jacobian tests, the hand-coded Jacobian is compared against a finite difference Jacobian from PETSC. The change in the weight function should not affect Jacobian tests. Am I misunderstanding something here?

What would be the best way to change the weight function?

Best regards
Andris

**Comment by GiudGiud** [comment, marked as answer]
Hello

My guess would first be that it s hardcoded some place. I see it in both the materials and the kernels involved
@hchen139 will have more insights

Guillaume

**Comment by andrisfreimanis** [comment]
Thanks, Guillaume. 

Took a closer look at the kernel and it is also hard coded there in nonlocal and nonlocal off-diagonal contributions. Changed those and the jacobian tests pass. 

Andris
